src/nix-channel/nix-channel.cc

```diff
@@ -165,7 +166,7 @@ static int main_nix_channel(int argc, char ** argv)
         // Figure out the name of the `.nix-channels' file to use
         auto home = getHome();
         channelsList = settings.useXDGBaseDirectories ? createNixStateDir() + "/channels" : home + "/.nix-channels";
```

Comment:
This line will also create the nix state dir, but arguably this is also unnecessary. The real point it should happen is Line 160 above.